modules/videoio/test/test_ffmpeg.cpp

```diff
@@ -476,6 +476,24 @@ static void ffmpeg_check_read_raw(VideoCapture& cap)
     EXPECT_EQ((size_t)37118, data.total());
 }
 
+TEST(videoio_ffmpeg, ffmpeg_check_extra_data)
+{
+    if (!videoio_registry::hasBackend(CAP_FFMPEG))
+        throw SkipTestException("FFmpeg backend was not found");
+
+    string video_file = findDataFile("video/big_buck_bunny.mp4");
+    VideoCapture cap;
+    EXPECT_NO_THROW(cap.open(video_file, CAP_FFMPEG, {
+        CAP_PROP_FORMAT, -1  // demux only
+        }));
+
+    Mat data;
+    cap.retrieve(data, 1);
```

Comment:
> 1

1. It is better to avoid magic numbers.
2. How users/application may know if this feature supported? This can be done through property, see my previous comment with:
    `cap.retrieve(codec_extradata_buffer, <video_codec_extradata_idx_property_value>);`
    Value in brackets should be fetched first through
    `int codec_extradata_idx = (int)cap.get(CAP_PROP_CODEC_EXTRADATA_INDEX);`
    and validated
    `if (codec_extradata_idx <= 0) bailout;`
    Retrieve call should go after that:
    `cap.retrieve(codec_extradata_buffer, codec_extradata_idx);`
3. Need to decide do we want to expose `code_extradata` for RAW streams only or for other cases too.

Reply:
> How users/application may know if this feature supported? This can be done through property, see my previous comment with:
> cap.retrieve(codec_extradata_buffer, <video_codec_extradata_idx_property_value>);
> Value in brackets should be fetched first through
> int codec_extradata_idx = (int)cap.get(CAP_PROP_CODEC_EXTRADATA_INDEX);
> and validated
> if (codec_extradata_idx <= 0) bailout;
> Retrieve call should go after that:
> cap.retrieve(codec_extradata_buffer, codec_extradata_idx);

Should the extraData field be used to return other info aswell in the future?   If so is it better to have a separate enum for the type of data
```
enum {
    CODEC_EXTRADATA = 0
}
```
or to include two properties in the existing enum as
```
CAP_PROP_CODEC_EXTRADATA_FLAG
CAP_PROP_CODEC_EXTRADATA_SUPPORTED
```

I'm not sure if it would then make sense to have something like
```
bool codec_extradata_supported = (bool)cap.get(CAP_PROP_CODEC_EXTRADATA_SUPPORTED);
if (!codec_extradata_supported ) bailout;
cap.retrieve(codec_extradata_buffer, CODEC_EXTRADATA (or CAP_PROP_CODEC_EXTRADATA_FLAG));
```

Alternatively we could stick with a single flag as
```
if (!cap.get(CAP_PROP_CODEC_EXTRADATA_INDEX)) bailout;
cap.retrieve(codec_extradata_buffer, CAP_PROP_CODEC_EXTRADATA_INDEX);
```

> Need to decide do we want to expose code_extradata for RAW streams only or for other cases too.

I can't think of a use case for this, but I also can't see any reason for not also exposing this capability when decoding.

Reply:
Single property should be enough.
"Support" check should look like `if (cap.get(CAP_PROP_CODEC_EXTRADATA_INDEX) <= 0) bailout;`. "0" parameter for `.retrieve()` is already reserved for the video data so it is considered as invalid for codec's extradata.

Direct using `CAP_PROP_CODEC_EXTRADATA_INDEX` as index is not enough for `.retrieve()` call.
It would be hard to avoid conflict with other backends or features (e.g., audio). Also there are different types/meaning of "property" and ".retrieve() index" - we don't want to mix them as we don't mix seconds vs meters. Please use returned value from .get() instead (result is backend-specific).

---

> other info aswell in the future

For other info we could add similar properties, e.g.

```
CAP_PROP_CAMERA_PINHOLE_INTRINSIC_MATRIX_INDEX = ...,  //!< .retrieve() index to fetch camera intrinsic matrix
CAP_PROP_CAMERA_PINHOLE_DISTORTION_PARAMETERS_INDEX = ...,  //!< .retrieve() index to fetch camera distortion parameters
```

(backends may support or not support them).


---

May be we could improve property naming:

- CAP_PROP_CODEC_EXTRADATA_INDEX
- CAP_PROP_CODEC_EXTRADATA_RETRIEVE_INDEX
- CAP_PROP_RETRIEVE_CODEC_EXTRADATA_INDEX
- CAP_PROP_RETRIEVE_INDEX_CODEC_EXTRADATA
- other suggestions

/cc @vpisarev @asmorkalov

Reply:
CAP_PROP_RETRIEVE_CODEC_EXTRADATA_FLAG?